### PR TITLE
media: platform: npcm750: update vcd and ece driver

### PR DESCRIPTION
**Pull request by maxdog988**
1. vcd/ece: move device initialization to probe function
2. vcd: introduce wait queue
3. vcd: add a new ioctl command for DE/HSYNC switch
4. vcd: configure default mode in dts.
5. vcd: remove unused struct.
6. vcd: enable interrupt before captrue and disable after captured.

### CODE REVIEW

File: drivers/media/platform/nuvoton/npcm750_vcd.c [hunk 1]
```diff
@@ -133,6 +138,64 @@
 #define  VCD_INTE_BSD3_IE	BIT(13)
 #define  VCD_INTE_VAL	(VCD_INTE_DONE_IE | VCD_INTE_IFOR_IE)
 
+#define VCD_RCHG	0x8028
+#define VCD_RCHG_TIM_PRSCL  GENMASK(12, 9)
+#define VCD_RCHG_IG_CHG2  GENMASK(8, 6)
+#define VCD_RCHG_IG_CHG1  GENMASK(5, 3)
+#define VCD_RCHG_IG_CHG0  GENMASK(2, 0)
+#define VCD_RCHG_TIM_PRSCL_OFFSET 9
```
Comment (avifishman) [inline drivers/media/platform/nuvoton/npcm750_vcd.c]: This is minor:
I don't see you are using those _OFFSET registers, but if you do want the put them before and then use them:
#define VCD_RCHG_TIM_PRSCL_OFFSET 9
#define VCD_RCHG_IG_CHG2_OFFSET 6
#define VCD_RCHG_IG_CHG1_OFFSET 3
#define VCD_RCHG_IG_CHG0_OFFSET 0
#define VCD_RCHG_TIM_PRSCL  GENMASK(12, VCD_RCHG_TIM_PRSCL_OFFSET)
#define VCD_RCHG_IG_CHG2  GENMASK(8, VCD_RCHG_IG_CHG2_OFFSET)
#define VCD_RCHG_IG_CHG1  GENMASK(5, VCD_RCHG_IG_CHG1_OFFSET)
#define VCD_RCHG_IG_CHG0  GENMASK(2, VCD_RCHG_IG_CHG0_OFFSET)

Comment (avifishman) [inline drivers/media/platform/nuvoton/npcm750_vcd.c]: This is minor:
I don't see you are using those _OFFSET registers, but if you do want the put them before and then use them:
#define VCD_RCHG_TIM_PRSCL_OFFSET 9
#define VCD_RCHG_IG_CHG2_OFFSET 6
#define VCD_RCHG_IG_CHG1_OFFSET 3
#define VCD_RCHG_IG_CHG0_OFFSET 0
#define VCD_RCHG_TIM_PRSCL GENMASK(12, VCD_RCHG_TIM_PRSCL_OFFSET)
#define VCD_RCHG_IG_CHG2 GENMASK(8, VCD_RCHG_IG_CHG2_OFFSET)
#define VCD_RCHG_IG_CHG1 GENMASK(5, VCD_RCHG_IG_CHG1_OFFSET)
#define VCD_RCHG_IG_CHG0 GENMASK(2, VCD_RCHG_IG_CHG0_OFFSET)

File: drivers/media/platform/nuvoton/npcm750_vcd.c [hunk 2]
```diff
@@ -542,11 +618,19 @@ static void npcm750_vcd_detect_video_mode(struct npcm750_vcd *vcd)
...
+		vcd->info.vbackporch = npcm750_vcd_verbp(vcd);
+	}
+
+	/* TODO: HSYNC mode: adjust delay when using internal gfx */
+	npcm750_vcd_dvod(vcd, vcd->info.hbackporch, vcd->info.vbackporch);
+
 	npcm750_vcd_set_linepitch(
 		vcd, vcd->info.hdisp * npcm750_vcd_get_bpp(vcd));
 	vcd->info.line_pitch = npcm750_vcd_get_linepitch(vcd);
 	npcm750_vcd_kvm_bw(vcd, vcd->info.pixelclk > 120000000UL);
```
Comment (avifishman) [inline drivers/media/platform/nuvoton/npcm750_vcd.c]: Add 120000000UL as a define since on next chip version it could be changed.

**Comment by maxdog988** [comment]
Thanks for your review

**Comment by avifishman** [comment]
Josheph,

You can push your changes directly to https://github.com/Nuvoton-Israel/linux/tree/Poleg-4.17.04-OpenBMC.

**Comment by maxdog988** [comment]
ERROR: Permission to Nuvoton-Israel/linux.git denied to maxdog988.
fatal: Could not read from remote repository.

**Comment by maxdog988** [comment]
Avi:

I don't have write access to push this change directly to Nuvoton-Israel/linux/tree/Poleg-4.17.04-OpenBMC..

**Comment by avifishman** [comment]
@maxdog988 
I changed your permission to be a contributor.
Do you suggest to add more people?

**Comment by maxdog988** [comment]
@avifishman 

Please add @stanleychuys for JTAG Mater driver.

Thanks

**Comment by avifishman** [comment]
Done

**Pull request closed without merging**